src/object/srv_obj.c

```diff
@@ -494,14 +494,42 @@ bulk_transfer_sgl(daos_handle_t ioh, crt_rpc_t *rpc, crt_bulk_t remote_bulk,
 int
 obj_bulk_transfer(crt_rpc_t *rpc, crt_bulk_op_t bulk_op, bool bulk_bind, crt_bulk_t *remote_bulks,
 		  uint64_t *remote_offs, uint8_t *skips, daos_handle_t ioh, d_sg_list_t **sgls,
-		  int sgl_nr, int bulk_nr, struct obj_bulk_args *p_arg)
+		  int sgl_nr, int bulk_nr, struct obj_bulk_args *p_arg, struct dtx_handle *dth,
+		  enum fi_leader fi_leader_value)
 {
 	struct obj_bulk_args	arg = { 0 };
 	int			i, rc, *status, ret;
 	int			skip_nr = 0;
 	bool			async = true;
 	uint64_t		time = daos_get_ntime();
 
+	// Fault injection for the case where the compound RPC body bulk transfer may throw a
+	// DER_RECONNECT. dth will be NULL in this test case.
+	if (dth == NULL && bulk_op == CRT_BULK_GET) {
+		if (DAOS_FAIL_CHECK(DAOS_CLIENT_UNREACHABLE_CPD_BODY) &&
```

Comment:
if ((DAOS_FAIL_CHECK_DAOS_CLIENT_UNREACHABLE) || DAOS_FAIL_CHECK(DAOS_CLIENT_UNREACHABLE_CPD_BODY)) && fi_leader_value == FOLLOWER) 

No need to have two blocks.  You can't set more than one fail loc at a time.

Reply:
another option would be to land this (into your branch)
https://github.com/daos-stack/daos/pull/16248/